src/Model/ASN.php

```diff
@@ -291,7 +275,7 @@ public function getName()
      *
      * @return $this
      */
-    public function setName($name)
+    public function setName($name): self
```

Comment:
Here and other model setters, can we define argument type too?

Reply:
It's impossible to do so without modifying generator because datatype is like this
```
public function setSubdivisions(\Fingerprint\ServerAPI\Model\Subdivision[] $subdivisions): self
```
This also applies for getters.

Problem is, you can't have data type in PHP like this `Subdivision[]` you can instead have data type like this `array`. To get get this we need to modify datatype in generator. But if you could suggest a better method let's discuss.

Reply:
I have a solution for such cases!

```
    public function {{setter}}({{^isListContainer}}{{^isMapContainer}}{{datatype}}{{/isMapContainer}}{{/isListContainer}}{{#isListContainer}}array{{/isListContainer}}{{#isMapContainer}}array{{/isMapContainer}} ${{name}}): self
```

Reply:
That generates another problem
```php
public function setLatitude(double $latitude): self {
```

PHP expects float instead of double, because there is no type such as double in PHP. Do you know a workaround for this or should I modify generate script to convert double types to float (I assume it'll take time)?

Reply:
`{{#isDouble}}{{/isDouble}}` should help

Reply:
It's done, can you resolve please?